js/app/build_plugins.ts

```diff
@@ -17,70 +17,47 @@ export function inject_ejs(): Plugin {
 	};
 }
 
-interface PatchDynamicImportOptionms {
-	mode: "cdn" | "local";
-	gradio_version: string;
-	cdn_url: string;
-}
-
-export function patch_dynamic_import({
-	mode,
-	gradio_version,
-	cdn_url
-}: PatchDynamicImportOptionms): Plugin {
-	return {
-		name: "patch-dynamic-import",
-		enforce: "post",
-		writeBundle(config, bundle) {
-			// const import_re = /import\(((?:'|")[\.\/a-zA-Z0-9]*(?:'|"))\)/g;
-			// const import_meta = `${"import"}.${"meta"}.${"url"}`;
-			// for (const file in bundle) {
-			// 	const chunk = bundle[file];
-			// 	if (chunk.type === "chunk") {
-			// 		if (chunk.code.indexOf("import(") > -1) {
-			// 			const fix_fn = `const VERSION_RE = new RegExp("${gradio_version}\/", "g");function import_fix(mod, base) {const url =  new URL(mod, base); return import(\`${cdn_url}\${url.pathname?.startsWith('/') ? url.pathname.substring(1).replace(VERSION_RE, "") : url.pathname.replace(VERSION_RE, "")}\`);}`;
-			// 			chunk.code =
-			// 				fix_fn +
-			// 				chunk.code.replace(import_re, `import_fix($1, ${import_meta})`);
-			// 			if (!config.dir) break;
-			// 			const output_location = join(config.dir, chunk.fileName);
-			// 			writeFileSync(output_location, chunk.code);
-			// 		}
-			// 	}
-			// }
-		}
-	};
-}
-
 export function generate_cdn_entry({
-	enable,
-	cdn_url
+	version,
+	cdn_base
 }: {
-	enable: boolean;
-	cdn_url: string;
+	version: string;
+	cdn_base: string;
 }): Plugin {
 	return {
 		name: "generate-cdn-entry",
 		enforce: "post",
 		writeBundle(config, bundle) {
-			if (!enable) return;
-
 			if (
 				!config.dir ||
 				!bundle["index.html"] ||
 				bundle["index.html"].type !== "asset"
 			)
 				return;
 
-			const tree = parse(bundle["index.html"].source as string);
-			const script =
-				Array.from(tree.querySelectorAll("script[type=module]")).find(
-					(node) => node.attributes.src?.startsWith(cdn_url)
-				)?.attributes.src || "";
+			const source = bundle["index.html"].source as string;
+			const tree = parse(source);
+
+			const script = Array.from(
+				tree.querySelectorAll("script[type=module]")
+			).find((node) => node.attributes.src?.includes("assets"));
 
 			const output_location = join(config.dir, "gradio.js");
 
-			writeFileSync(output_location, make_entry(script));
+			writeFileSync(output_location, make_entry(script?.attributes.src || ""));
+
+			if (!script) return;
+
+			const transformed_html =
+				(bundle["index.html"].source as string).substring(0, script?.range[0]) +
+				`<script type="module" crossorigin src="${cdn_base}/${version}/gradio.js"></script>` +
+				(bundle["index.html"].source as string).substring(
+					script?.range[1],
+					source.length
+				);
+
+			const share_html_location = join(config.dir, "share.html");
+			writeFileSync(share_html_location, transformed_html);
```

Comment:
This is for the `share.html` rather than copying it over like we used to, we created it directly from the `index.html`. What we are doing here is prepending the `cdn_base` to the scripts src. This is the only time we need to do this.